### ISSUE TITLE
Scipy dblquad/tplquad spuriously drops epsilon for inner integration (Trac #1572)

### ISSUE THREAD

**Issue by scipy-gitbot** [scipy/scipy#2097]
_Original ticket http://projects.scipy.org/scipy/ticket/1572 on 2011-12-02 by trac user trobinso, assigned to unknown._

From devel _scipy/integrate/quadpack.py_, the relative and absolute epsilon values appear to be getting dropped on the floor for the inner integration step:

```
def _infunc(x,func,gfun,hfun,more_args):
    a = gfun(x)
    b = hfun(x)
    myargs = (x,) + more_args
    return quad(func,a,b,args=myargs)[0] # <-- Here

def dblquad(func, a, b, gfun, hfun, args=(), epsabs=1.49e-8, epsrel=1.49e-8):
    ...
    return quad(_infunc,a,b,(func,gfun,hfun,args),epsabs=epsabs,epsrel=epsrel)

def _infunc2(y,x,func,qfun,rfun,more_args):
    a2 = qfun(x,y)
    b2 = rfun(x,y)
    myargs = (y,x) + more_args
    return quad(func,a2,b2,args=myargs)[0] # <-- And here

def tplquad(func, a, b, gfun, hfun, qfun, rfun, args=(), epsabs=1.49e-8,
            epsrel=1.49e-8):
    ...
    return dblquad(_infunc2,a,b,gfun,hfun,(func,qfun,rfun,args),epsabs=epsabs,epsrel=epsrel)
```

This can be fixed with a quick refactoring of the inner operation. For example:

```
def _infunc(x,func,gfun,hfun,epsabs,epsrel,more_args):
    a = gfun(x)
    b = hfun(x)
    myargs = (x,) + more_args
    return quad(func,a,b,args=myargs,epsabs=epsabs,epsrel=epsrel)[0]

def dblquad(func, a, b, gfun, hfun, args=(), epsabs=1.49e-8, epsrel=1.49e-8):
    ...
    return quad(_infunc,a,b,(func,gfun,hfun,epsabs,epsrel,args),epsabs=epsabs,epsrel=epsrel)
```

I've made these changes to my local version, since I'm setting a large epsilon for a faster approximation. 

Is there a good reason why this information is being removed during the handoff?

**Comment by adriesse** [comment]
Just a comment that I came across the problem too, trying to speed up my calculations, but not succeeding.
scipy version 0.18.1

**Comment by pv** [comment]
probably was fixed in 0.19.0 last year

**Comment by pv** [comment]
Indeed was fixed.
